org.bardsoftware.eclipsito/src/org/bardsoftware/impl/eclipsito/BootImpl.java

```diff
@@ -35,6 +39,42 @@ public void run(String application, File modulesDir, String descriptorPattern, L
       return plugins;
     }
 
+  public File getVersionDir(List<File> modulesDir) {
+    String version = null;
+    String pluginsPath = null;
+    if (modulesDir != null) {
+      if (modulesDir.isEmpty()) {
+        return null;
```

Comment:
You return null but you never check in the caller method if the returned value is null or not